packages/dds/merge-tree/src/client.ts

```diff
@@ -542,22 +543,21 @@ export class Client {
             }
 
             if (invalidPositions.length > 0) {
-                this.logger.sendErrorEvent({
-                    currentSeq: this.getCurrentSeq(),
-                    end,
-                    eventName: "InvalidOpRange",
-                    invalidPositions: invalidPositions.toString(),
-                    length,
-                    opPos1: op.pos1,
-                    opPos1Relative: op.relativePos1 !== undefined,
-                    opPos2: op.pos2,
-                    opPos2Relative: op.relativePos2 !== undefined,
-                    opRefSeq: clientArgs.referenceSequenceNumber,
-                    opSeq: clientArgs.sequenceNumber,
-                    opType: op.type,
-                    start,
-                });
-                return undefined;
+                throw new LoggingError(
+                    "RangeOutOfBounds",
+                    {
+                        UsageError: true,
```

Comment:
I'm a bit confused by UsageError = true. As far as I can see, this API is called in op processing. I'd assume that if we allowed op to go out, it's not usage issue, it's DDS error, either in initial validation or something else (or maybe error in op stream).
And assuming that's the case, what's the point of adding all that data into logging? We either have no access to file (and then it's pretty useless), or we have access to a file and it repros on every run, so data is available.

Can you please clarify?

Reply:
we only throw this error for local operation; we check above that the client id == the local client id